src/sentry/models/team.py

```diff
@@ -315,9 +318,19 @@ def get_projects(self):
 
         return Project.objects.get_for_team_ids({self.id})
 
+    def outbox_for_update(self) -> RegionOutbox:
+        return RegionOutbox(
+            shard_scope=OutboxScope.TEAM_SCOPE,
+            shard_identifier=self.organization_id,
+            category=OutboxCategory.TEAM_UPDATE,
+            object_identifier=self.id,
+        )
+
     def delete(self, **kwargs):
         from sentry.models import ExternalActor
 
         # There is no foreign key relationship so we have to manually delete the ExternalActors
-        ExternalActor.objects.filter(actor_id=self.actor_id).delete()
-        return super().delete(**kwargs)
+        with transaction.atomic(), in_test_psql_role_override("postgres"):
```

Comment:
Are we blocking delete on Team?

Reply:
No. Delete on team should not be blocked. This new piece of code creates a RegionOutbox when deleting a Team object.